### PR TITLE
Add support to read varbinary column from Parquet fixed length byte array

### PR DESCRIPTION
**Pull request by majetideepak**
Resolves: https://github.com/facebookincubator/velox/issues/9757

### CODE REVIEW

**Review by yma11 — APPROVED** [review]
LGTM. Thanks for the fix!

File: velox/dwio/parquet/reader/StringDecoder.h [hunk 1]
```diff
@@ -85,9 +92,16 @@ class StringDecoder {
...
+
+  folly::StringPiece readFixedString() {
+    bufferStart_ += length_;
+    return folly::StringPiece(bufferStart_ - length_, length_);
+  }
+
   const char* bufferStart_;
   const char* bufferEnd_;
   const char* const lastSafeWord_;
+  int length_;
```
Comment (Yuhta) [inline velox/dwio/parquet/reader/StringDecoder.h:104]: nit: `const`

**Review by Yuhta — APPROVED** [review]

**Review by yingsu00 — COMMENTED** [review]
LGTM just some nits.

File: velox/dwio/parquet/reader/StringDecoder.h [hunk 2]
```diff
@@ -23,8 +23,14 @@ class StringDecoder {
   StringDecoder(const char* start, const char* end)
       : bufferStart_(start),
         bufferEnd_(end),
+        lastSafeWord_(end - simd::kPadding),
+        length_(-1) {}
 
-        lastSafeWord_(end - simd::kPadding) {}
+  StringDecoder(const char* start, const char* end, int length)
```
[the remark this answers is not in the document]
Comment (majetideepak) [reply velox/dwio/parquet/reader/StringDecoder.h]: fixed

File: velox/dwio/parquet/reader/StringDecoder.h [hunk 3]
```diff
@@ -85,9 +92,16 @@ class StringDecoder {
...
+
+  folly::StringPiece readFixedString() {
+    bufferStart_ += length_;
+    return folly::StringPiece(bufferStart_ - length_, length_);
+  }
+
   const char* bufferStart_;
   const char* bufferEnd_;
   const char* const lastSafeWord_;
+  const int length_;
```
[the remark this answers is not in the document]
Comment (majetideepak) [reply velox/dwio/parquet/reader/StringDecoder.h]: fixed

**Review by yingsu00 — APPROVED** [review]

**Comment by mbasmanova** [comment]
Seeing test failure:

```
[ RUN      ] ParquetReaderTest.readVarbinaryFromFLBA

E0524 07:50:27.656497  8491 Exceptions.h:67] Line: fbcode/velox/dwio/parquet/reader/ParquetReader.cpp:136, Function:ReaderBase, Expression: fileLength_ > 0 (0 vs. 0) Parquet file is empty, Source: RUNTIME, ErrorCode: INVALID_STATE
terminate called after throwing an instance of 'facebook::velox::VeloxRuntimeError'
  what():  Exception: VeloxRuntimeError
Error Source: RUNTIME
Error Code: INVALID_STATE
Reason: (0 vs. 0) Parquet file is empty
Retriable: False
Expression: fileLength_ > 0
Function: ReaderBase
File: fbcode/velox/dwio/parquet/reader/ParquetReader.cpp
Line: 136
```

**Comment by majetideepak** [comment]
> Parquet file is empty

@mbasmanova, The CI test passed. The test passes locally as well. The file being empty is strange since I can that the file is now empty in this PR. Likely the file did not download correctly in the Meta test infrastructure? Can you please check again?

**Comment by mbasmanova** [comment]
@majetideepak Deepak, I'm seeing that after importing this PR, the new file varbinary_flba.parquet is empty. I'll need some time to figure out why this is happening and how to fix that.

**Pull request closed without merging**